gce/survival-training/README.md

```diff
@@ -0,0 +1,93 @@
+
+
+### Custom images
+
+Your training job may require specific versions of libraries, like [tensorflow](https://www.tensorflow.org/) or [scikit-learn](http://scikit-learn.org). Additionally, in order to take advantage of instances with attached GPUs, you will have to have hardware-specific drivers installed on the instance. [Custom images](https://cloud.google.com/compute/docs/images#custom_images) are a convenient solution to this problem of environment reproduction within a GCE virtual machine.
+
+If you do have requirements (in addition to simply the OS) in your training environments, it is highly recommended that you create a VM image that you can use to spawn new instances without having to go through the tedious configuration process every time. The examples following this strategy section will demonstrate the creation of such images so, if this seems daunting, don't worry.
+
+
+### Trainer CLI
+
+Our strategy will have the GCE instance automatically kick off your training job on startup. As part of this kick off, it will have to provide the job with the appropriate parameters -- how many steps to train for, how often to checkpoint the model, which data to train the model with, what hyperparameters should be used in training the model, etc.
+
+It is helpful, if you intend to do this more than once, to assert a common semantical system across all your trainers. We provide a [boilerplate CLI](./dummy/train.py) in this repo. This interface is very similar to that provided by [TensorFlow estimators](https://www.tensorflow.org/programmers_guide/estimators), but can be used even with other frameworks. In the sections that follow, we will provide you with both TensorFlow and with scikit-learn examples that demonstrate its use.
+
+
+### Cloud Source Repositories
+
+We need a mechanism by which we can get our trainer code into the GCE instance. There are several solutions available to us:
+
+1. We could simply store the trainer application on the custom image from which we create the GCE instance.
+
+1. We could store the code in a [GCS](https://cloud.google.com/storage/) bucket and have the startup script download it to the instance before doing anything else.
+
+1. We could store the code in a [Cloud Source Repository](https://cloud.google.com/source-repositories/) and have the startup script clone it to the instance before doing anything else.
+
+In this guide, in the interests of flexibility, we will go with the Cloud Source Repository option. The reason this is attractive is that it lets you apply `git` semantics to your training jobs -- for example, you can [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) different model architectures and specify them through the instance metadata. Or you could just use the trainer on some special branch of the repository which, again, would be specified at deployment time using instance metadata. This setup is much more powerful than the others.
```

Comment:
> In this guide, in the interests of flexibility, we will go with the Cloud Source Repository option. The reason this is attractive is that it lets you apply `git` semantics to your training jobs -- for example, you can [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) different model architectures and specify them through the instance metadata. Or you could just use the trainer on some special branch of the repository which, again, would be specified at deployment time using instance metadata. This setup is much more powerful than the others.

In this guide we will use the Cloud Source Repository option, because it is more flexible and powerful than the other options. Using Cloud Source Repository lets you apply `git` semantics to your training jobs. For example, if you [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) different model architectures or use your trainer only in a specific branch of the repository, your tags and branch selections would be specified at deployment time using instance metadata.